### PR TITLE
Added TLS handshake to RPCDaemon

### PR DESCRIPTION
**Pull request by Giulio2002**

### CODE REVIEW

**Comment by mandrigin** [comment]
@Giulio2002 looks great! 

(1) technically, does that mean that if we want to run multiple RPC daemons we will need to securely store the key file on each of them?

(2) also, regarding the secure defaults. I would much prefer inverted flags like `--notls` so the TLS is there by default.

File: ethdb/remote/remotedbserver/server.go [hunk 1]
```diff
@@ -27,6 +28,58 @@ type KvServer struct {
 	kv ethdb.KV
 }
 
+func StartGrpcWithTLS(kv ethdb.KV, eth core.Backend, addr string, certFile string, keyFile string) {
```
Comment (mandrigin) [inline ethdb/remote/remotedbserver/server.go:31]: this method is a bit too much copy-paste from `StartGrpc` and it differs only in in a couple of places.
can you somehow generalize that?

I see that there are just 2 differences:
(1) log message
(2) credentials passing

**Comment by mandrigin** [comment]
(3) we might want to use mutual TLS authentication, (so we both validate client and server). The example is here: https://github.com/grpc/grpc-go/issues/403 but this can be moved to the next PR.

**Comment by Giulio2002** [comment]
@mandrigin I finished the cleanup requested, I will do mutual authentication in another PR

**Comment by AlexeyAkhunov** [comment]
@Giulio2002 You can also do some minification in the `kv_remote.go` file, to make functions `OpenWithTls` and `Open` to not be almost copies of each other

**Comment by mandrigin** [comment]
@Giulio2002 looks much better! one more small suggestion and I think we can merge it

**Comment by Giulio2002** [comment]
Ok, done

File: cmd/utils/flags.go [hunk 2]
```diff
@@ -547,6 +547,20 @@ var (
 		Usage: "API's offered over the HTTP-RPC interface",
 		Value: "",
 	}
+	NoTLSFlag = cli.BoolFlag{
```
Comment (AlexeyAkhunov) [inline cmd/utils/flags.go:550]: Please replace with `BoolTFlag`, so it it will be true by default for now

**Pull request merged**